File: Cap 09/Topico 9.3/Topico 9.3.1 a 9.3.4 juntos/regmatrix.dart
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topico 9.3.1 ate 9.3.4
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ
 * 
 */
 
import 'dart:io';

// Nao e possivel em Dart definir o tipo BIMESTRE

class CAD_ALUNO {
  String NOME = '';
  String TURMA = '';
  int SALA = 0;
  List<double> NOTAS = [];
}

var ALUNO = CAD_ALUNO();
var I;

void main() {

  print('REGISTRO ESCOLAR - SIMPLES (1 ALUNO)\n');
  stdout.write('Entre o nome ......: ');
  ALUNO.NOME = stdin.readLineSync()!;
  stdout.write('Entre a turma .....: ');
  ALUNO.TURMA = stdin.readLineSync()!;
  stdout.write('Entre a sala ......: ');
  ALUNO.SALA = int.parse(stdin.readLineSync()!);
  ALUNO.NOTAS = [];
  for (I = 0; I <= 3; I++) {
    stdout.write('Entre a ${I + 1}a. nota ..: ');
    ALUNO.NOTAS.add(double.parse(stdin.readLineSync()!));
  }

  print('\nDADOS DO ALUNO\n');
  print('Nome ..............: ${ALUNO.NOME}');
  print('Turma .............: ${ALUNO.TURMA}');
  print('Sala ..............: ${ALUNO.SALA.toString().padLeft(4)}');
  for (I = 0; I <= 3; I++) {
    print('Nota ${I + 1} ............: ${ALUNO.NOTAS[I].toStringAsFixed(1).padLeft(4)}');
  }
  
}

File: Cap 09/Topico 9.3/Topico 9.3.1 a 9.3.4 juntos/regmatrix.cs
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topico 9.3.1 ate 9.3.4
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ
*/

using System;

// Nao e possivel em C# definir o tipo BIMESTRE

public struct CAD_ALUNO
{
  public string NOME;
  public char TURMA;
  public int SALA;
  public float[] NOTAS;
}

public class Program
{
  
  static CAD_ALUNO ALUNO;
  static int I;

  public static void Main()
  {
    
    Console.WriteLine("REGISTRO ESCOLAR - SIMPLES (1 ALUNO)\n");
    Console.Write("Entre o nome ......: ");
    ALUNO.NOME = Console.ReadLine();
    Console.Write("Entre a turma .....: ");
    ALUNO.TURMA = Console.ReadLine()[0];
    Console.Write("Entre a sala ......: ");
    ALUNO.SALA = int.Parse(Console.ReadLine());
    ALUNO.NOTAS = new float[4];
    for (I = 0; I <= 3; I++)
    {
      Console.Write($"Entre a {I + 1}a. nota ..: ");
      ALUNO.NOTAS[I] = float.Parse(Console.ReadLine());
    }

    Console.WriteLine("\nDADOS DO ALUNO\n");
    Console.WriteLine($"Nome ..............: {ALUNO.NOME}");
    Console.WriteLine($"Turma .............: {ALUNO.TURMA}");
    Console.WriteLine($"Sala ..............: {ALUNO.SALA,4}");
    for (I = 0; I <= 3; I++)
    {
      Console.WriteLine($"Nota {I + 1} ............: {ALUNO.NOTAS[I],4:F1}");
    }
    
  }
  
}
